File: src/main/java/com/refinedmods/refinedstorage/container/factory/WirelessCraftingMonitorContainerFactory.java
package com.refinedmods.refinedstorage.container.factory;

import com.refinedmods.refinedstorage.RSContainerMenus;
import com.refinedmods.refinedstorage.container.CraftingMonitorContainerMenu;
import com.refinedmods.refinedstorage.inventory.player.PlayerSlot;
import com.refinedmods.refinedstorage.blockentity.craftingmonitor.WirelessCraftingMonitor;
import net.minecraft.network.FriendlyByteBuf;
import net.minecraft.world.entity.player.Inventory;
import net.minecraft.world.item.ItemStack;
import net.minecraftforge.network.IContainerFactory;

public class WirelessCraftingMonitorContainerFactory implements IContainerFactory<CraftingMonitorContainerMenu> {
    @Override
    public CraftingMonitorContainerMenu create(int windowId, Inventory inv, FriendlyByteBuf data) {

        PlayerSlot slot = new PlayerSlot(data);

        ItemStack stack = slot.getStackFromSlot(inv.player);

        WirelessCraftingMonitor wirelessCraftingMonitor = new WirelessCraftingMonitor(stack, null, slot);

        return new CraftingMonitorContainerMenu(RSContainerMenus.WIRELESS_CRAFTING_MONITOR.get(), wirelessCraftingMonitor, null, inv.player, windowId);
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/container/factory/WirelessCraftingMonitorContainerFactory.kt
package com.refinedmods.refinedstorage.container.factory

import com.refinedmods.refinedstorage.RSContainers
import com.refinedmods.refinedstorage.api.network.grid.IGrid.slotId
import com.refinedmods.refinedstorage.container.CraftingMonitorContainer
import com.refinedmods.refinedstorage.tile.craftingmonitor.WirelessCraftingMonitor
import net.minecraft.entity.player.PlayerInventory
import net.minecraft.item.ItemStack
import net.minecraft.network.PacketByteBuf
import net.minecraftforge.fml.network.IContainerFactory

class WirelessCraftingMonitorContainerFactory : IContainerFactory<CraftingMonitorContainer?> {
    fun create(windowId: Int, inv: PlayerInventory, data: PacketByteBuf): CraftingMonitorContainer {
        val slotId: Int = data.readInt()
        val stack: ItemStack = inv.getStackInSlot(slotId)
        val wirelessCraftingMonitor = WirelessCraftingMonitor(stack, null, slotId)
        return CraftingMonitorContainer(RSContainers.WIRELESS_CRAFTING_MONITOR, wirelessCraftingMonitor, null, inv.player, windowId)
    }
}